edkrepo/commands/sync_command.py

```diff
@@ -168,12 +172,19 @@ def run_command(self, args, config):
         global_manifest_path = get_manifest_repo_path(manifest_repo, config)
         for repo_to_sync in repo_sources_to_sync:
             local_repo_path = os.path.join(workspace_path, repo_to_sync.root)
+            repo = Repo(local_repo_path)
+            if not args.fetch:
+                ui_functions.print_info_msg(humble.SYNCING.format(repo_to_sync.root, repo.active_branch), header = False)
+            else:
+                ui_functions.print_info_msg(humble.FETCHING.format(repo_to_sync.root, repo.active_branch), header = False)
+
             # Update any hooks
             if global_manifest_directory is not None:
                 update_hooks(hooks_add, hooks_update, hooks_uninstall, local_repo_path, repo_to_sync, config, global_manifest_directory)
-            repo = Repo(local_repo_path)
-            #Fetch notes
+
             repo.remotes.origin.fetch("refs/notes/*:refs/notes/*")
+            if args.tags:
+                repo.git.execute(['git', 'fetch', '--tags'])
```

Comment:
may want to put "git fetch --tags" into a try/except block.  
As is, if this line fails, the whole edkrepo sync commands stops, but we might be able to reduce it to a warning.

For example, if a user local tag conflicts with remote tag of the same name, GitCommandError exception occurs.
"raise GitCommandError(command, status, stderr_value, stdout_value)"
"! [rejected] tagname -> tagname (would clobber existing tag)"
"The git command: git fetch --tags failed to complete successfully with the following errors."